activemodel/test/cases/type/date_time_test.rb

```diff
@@ -25,6 +25,16 @@ def test_string_to_time_with_timezone
         end
       end
 
+      def test_hash_to_time
+        type = Type::DateTime.new
+        assert_equal ::Time.utc(2018, 10, 15, 0, 0, 0), type.cast(1 => 2018, 2 => 10, 3 => 15)
+      end
+
+      def test_hash_with_wrong_keys
+        type = Type::DateTime.new
+        assert_raises(ArgumentError) { type.cast(a: 1) }
```

Comment:
This would pass regardless of your change. I would consider asserting the error message like this:


```ruby
error = assert_raises(ArgumentError) { type.cast(a: 1) }
assert_equal "Some message", error.message
```